client.go

```diff
@@ -187,31 +185,39 @@ func (c *Client) Put(ctx context.Context, url string, contentType string, body i
 	return c.Do(ctx, req)
 }
 
-// PostForm calls Post with the appropriate form content-type
+// PostForm calls Post with the appropriate form content-type.
 func (c *Client) PostForm(ctx context.Context, uri string, data url.Values) (*http.Response, error) {
 	return c.Post(ctx, uri, "application/x-www-form-urlencoded", strings.NewReader(data.Encode()))
 }
 
-func (c *Client) backoff(ctx context.Context, doer func(context.Context, *http.Client, *http.Request) (*http.Response, error), retryErr error, client *http.Client, req *http.Request) (resp *http.Response, err error) {
-	if c.GetMaxRetries() < 1 {
-		return nil, retryErr
-	}
-	for attempt := 1; attempt <= c.GetMaxRetries(); attempt++ {
-		// ensure that the context is not cancelled before iterating
-		if ctx.Err() != nil {
+func (c *Client) backoff(
+	ctx context.Context,
+	doer func(context.Context, *http.Client, *http.Request) (*http.Response, error),
```

Comment:
nit picky comment but you could create an alias type for the doer func and update this func to take that type instead of declaring the func signature in line.  Makes the back func signature more concise/readable.